core/store/orm/orm.go

```diff
@@ -613,6 +622,22 @@ func (orm *ORM) UpdateTx(
 	return orm.DB.Save(tx).Error
 }
 
+// FindLaterConfirmedTx finds the earliest confirmed Ethereum transaction
+// that has the same sender and a higher nonce.
+func (orm *ORM) FindLaterConfirmedTx(tx *models.Tx) (*models.Tx, error) {
+	later := models.Tx{}
+	rval := orm.DB.Order("nonce asc").
+		Where("\"confirmed\" = true AND \"from\" = ? AND \"nonce\" > ?", tx.From, tx.Nonce).
+		First(&later)
+
+	removed, err := removeRecordNotFound(rval)
+	if removed {
+		return nil, err
+	}
```

Comment:
Is it common practice in the code-base to use 'nil' to signal a record not being found?

Reply:
I think so. Specifically, we do it when we don't want to raise an error because a record isn't found. The db typically returns an error, but that's an expected outcome for us, not necessarily an error.